lib/src/components/mixins/collidable.dart

```diff
@@ -4,6 +4,26 @@ import '../../geometry/rectangle.dart';
 import 'hitbox.dart';
 
 mixin Collidable on Hitbox {
+  /// Set whether [Collidable] is active
+  /// If [activeCollidable] is set to true, collision with other [Collidable] will be actively checked in [collisionDetection]
+  /// [Collidable] whose [activeCollidable] is set to false will not actively check collisions with other [Collidable],
+  /// so as to prevent collision detection between stationary [Collidable] and other stationary [Collidable], so as to improve the performance of [collisionDetection]
+  bool _activeCollidable = null;
```

Comment:
Maybe it is cleaner to use a Enum value for this instead of null and check against that?

This will make it easier to maintain for null-safety but also wont create unexpected behaviour as you can just compare against enum values to get the bool value for the getter.

```dart
enum IsCollidable {
  isNotDefined,
  isActive,
  isNotActive,
}
```

That way we can keep the checks in that compare with `null`(but instead of null the appropriate enum value).

Just an idea, I will refer/follow whatever @spydon will suggest :D

Reply:
I agree that it would be better if spydon is true, and I will update these codes if [Active] is really needed

> https://github.com/flame-engine/flame/pull/690#pullrequestreview-600680436

>https://github.com/flame-engine/flame/pull/690#issuecomment-787997490